### ISSUE TITLE
Voice enhancements

### ISSUE THREAD

**Issue by btopro** [elmsln/elmsln#1151]
See if the browser can detect and access accessories plugged into the device. For example, if you could activate when the user presses the microphone button on their earbuds that'd be pretty sick.

Also (much more realistic) we should be using the shut-down hook currently used in debug mode to list what's being said as a way of doing more accurate detection during background noise / free form speech. For example, if we detect `So you see, what I can do next is say something like elms next page and it'll go to the next page` and we have a trigger for `(phrase) next page` then we should be able to regex this out on exit of the processing and then match the function.  This is a fallback but would also allow for natural forms of speech. We could drop matches for optional phrases appearing in `( )` but require the use of the trigger word in order to do this matching so that you aren't just saying 'hey i really love the next page' and have it trigger.

This also puts things more in line with the 'ok google' style voice command recognition stuff of today.

**Comment by btopro** [comment]
unreasonable and we already switched things to "Hey elms" because O.K. is hard to differentiate from Okay and OK and ok and OH KAY
